### DISCUSSION TITLE
Example programs for Autodiff with GPU/CUDA?

### DISCUSSION THREAD

**Discussion by jgjonsson** [autodiff/autodiff#333] [Q&A]
The feature for [GPU support](https://github.com/autodiff/autodiff/pull/309) added in 1.1.0 looks very interesting.

Are there any plans to add examples to [tutorial ](https://autodiff.github.io/tutorials/)section? I'm thinking some minimal *.cu file calculating a derivative or gradient, that can be compiled to a program as is with nvcc. 
Or if someone has any such file they can share on the side?  

I realize this is an experimental new feature that only covers forward mode right now. I'm nonetheless interested in trying it out for a university project. I could possibly find one via the tests under tests/forward how it works, but thought a minimal example file would be a much faster starting point.

**Comment by allanleal** [comment]
Hi @jgjonsson - I think this is a great idea. Thanks for suggesting it. The GPU support in autodiff was implemented by @JieRen98. 

Could you two or one of you please contribute to this?

**Reply by JieRen98** [reply]
Yes sure, I will do it this weekend.

**Reply by JieRen98** [reply]
If you think that is urgent I can do it earlier.

**Reply by jgjonsson** [reply]
Not urgent. This weekend would be great, thank you!

**Reply by allanleal** [reply]
Hi @JieRen98 , very kind of you. Many thanks for supporting this project.

**Reply by jgjonsson** [reply]
Hi @JieRen98! Was there something added yet? (Not sure if it'll be on tutorial page or repo examples directory.) 

Also mentioning that I'm in particular interested in _Derivatives of a single-variable function_, and _Gradient vector of a scalar function_. Is the gradient in forward mode currently supported?

**Reply by kursatyurt** [reply]
@JieRen98 @jgjonsson Is there any update?

**Comment by EthanHarness** [comment]
@JieRen98 @jgjonsson 

I wanted to follow up on this and also ask what CUDA versions were supported. Additionally, I wanted to ask if reverse has been supported yet since I found this comment on the PR conversation: https://github.com/autodiff/autodiff/pull/309#issuecomment-1880149001
